managed/services/management/dbaas/pxc_cluster_service.go

```diff
@@ -78,26 +78,24 @@ func (s PXCClustersService) GetPXCClusterCredentials(ctx context.Context, req *d
 	if err != nil {
 		return nil, err
 	}
-
-	in := &dbaascontrollerv1beta1.GetPXCClusterCredentialsRequest{
-		KubeAuth: &dbaascontrollerv1beta1.KubeAuth{
-			Kubeconfig: kubernetesCluster.KubeConfig,
-		},
-		Name: req.Name,
+	if err := s.kubernetesClient.SetKubeconfig(kubernetesCluster.KubeConfig); err != nil {
+		return nil, errors.Wrap(err, "failed creating kubernetes client")
 	}
-
-	cluster, err := s.controllerClient.GetPXCClusterCredentials(ctx, in)
+	dbCluster, err := s.kubernetesClient.GetDatabaseCluster(ctx, req.Name)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed getting database cluster")
+	}
+	secret, err := s.kubernetesClient.GetSecret(ctx, fmt.Sprintf(pxcSecretNameTmpl, req.Name))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed getting secret")
 	}
 
-	_ = kubernetesCluster
 	resp := dbaasv1beta1.GetPXCClusterCredentialsResponse{
 		ConnectionCredentials: &dbaasv1beta1.PXCClusterConnectionCredentials{
-			Username: cluster.Credentials.Username,
-			Password: cluster.Credentials.Password,
-			Host:     cluster.Credentials.Host,
-			Port:     cluster.Credentials.Port,
+			Username: "root",
+			Password: string(secret.Data["root"]),
+			Host:     dbCluster.Status.Host,
+			Port:     3306,
```

Comment:
This is OK for now, but if we are allowing users to pass MySQL configuration options, like in `my.cnf`, there is a `port` option that changes the default port.

Reply:
Thanks. It's a great point. I'll create a Jira ticket for that.